rasa/utils/train_utils.py

```diff
@@ -0,0 +1,928 @@
+from collections import namedtuple
+import logging
+import numpy as np
+from tqdm import tqdm
+from sklearn.model_selection import train_test_split
+import tensorflow as tf
+from tensor2tensor.models.transformer import (
+    transformer_base,
+    transformer_prepare_encoder,
+    transformer_encoder,
+)
+from tensor2tensor.layers.common_attention import large_compatible_negative
+from rasa.utils.common import is_logging_disabled
+import typing
+from typing import List, Optional, Text, Dict, Tuple, Union, Generator, Callable
+
+if typing.TYPE_CHECKING:
+    from tensor2tensor.utils.hparam import HParams
+
+# avoid warning println on contrib import - remove for tf 2
+tf.contrib._warning = None
+logger = logging.getLogger(__name__)
+
+# namedtuple for all tf session related data
+SessionData = namedtuple("SessionData", ("X", "Y", "labels"))
```

Comment:
what is the difference between `Y` and `labels`? I think this is named a bit odd, as `y` is also regularly called "labels"

Reply:
`Y` is features for `labels` which in this case are ids, renamed to `label_ids`